### PR TITLE
Fix task pre-processor in tasks

### PR DESCRIPTION
**Pull request by sampathweb**
When a user passes `preprocessor` to a `from_preset` constructor via kwargs, it fails with "got multiple values for keyword argument 'preprocessor'".  This is because `from_preset` adds `preprocessor` parameter regardless of whether has also provided it.  With this fix, we only add the `preprocessor` when the user has not provided the input.

Also, adds a validation to verify that user doesn't provide a `backbone` to the `from_preset` method since that's obtained from the preset directly.

```
# Repro steps
preprocessor = keras_nlp.models.DistilBertPreprocessor.from_preset(
    "distil_bert_base_en_uncased",
    sequence_length=SEQ_LENGTH,
)

# Pretrained classifier -  Errors out with duplicate `preprocessor` arg
classifier = keras_nlp.models.DistilBertClassifier.from_preset(
    "distil_bert_base_en_uncased",
    num_classes=2,
    activation=None,
    preprocessor=preprocessor,
)
```

### CODE REVIEW

File: keras_nlp/models/task.py [hunk 1]
```diff
@@ -191,7 +191,16 @@ def from_preset(
                 preset,
                 config_file="tokenizer.json",
             )
-            preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)
+            if "backbone" in kwargs:
+                raise ValueError(
+                    "`backbone` weights are loaded from the "
```
Comment (mattdangerw) [inline keras_nlp/models/task.py:196]: I would reword this so we give the user an action they can do to remedy this, which is pretty simple they should call the default constructor and not `from_preset`. Something like...

```python
"You cannot pass a `backbone` argument to the `from_preset` method. "
f"Instead, call the {cls.__name__} default constructor with `backbone` argument. "
f"Received: backbone={backbone}"
```

File: keras_nlp/models/task.py [hunk 2]
```diff
@@ -191,7 +191,16 @@ def from_preset(
                 preset,
                 config_file="tokenizer.json",
             )
-            preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)
+            if "backbone" in kwargs:
```
Comment (mattdangerw) [inline keras_nlp/models/task.py:194]: Let's rework the structure of this so we don't create a useless tokenizer if a preprocessor object is getting passed in. And raise all errors first, before downloading files.

```python
if preset_cls == cls.backbone_cls:
    # Load backbone.
    if "backbone" in kwargs:
        raise ...
    backbone = load_from_preset(....

    # Load preprocessor.
    if "preprocessor" in kwargs:
        preprocessor = kwargs.pop("preprocessor")
    else:
        tokenizer = load_from_preset(...
        preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)

    # Create task.
    return cls(backbone=backbone, preprocessor=preprocessor, **kwargs)
```

**Review by mattdangerw — APPROVED** [review]
Thank you! Couple comments.

File: keras_nlp/models/task.py (hunk 1 above)
Comment (sampathweb) [reply keras_nlp/models/task.py]: Thanks for rewording it. Done.

File: keras_nlp/models/task.py (hunk 2 above)
Comment (sampathweb) [reply keras_nlp/models/task.py:190]: Makes sense. Done.

File: keras_nlp/models/task.py [hunk 3]
```diff
@@ -187,11 +187,21 @@ def from_preset(
...
-            tokenizer = load_from_preset(
-                preset,
-                config_file="tokenizer.json",
-            )
-            preprocessor = cls.preprocessor_cls(tokenizer=tokenizer)
+            if "backbone" in kwargs:
+                raise ValueError(
+                    "You cannot pass a `backbone` argument to the "
+                    f"`from_preset` method. Instead, call the {cls.__name__} "
+                    "default constructor with `backbone` argument. "
```
Comment (mattdangerw) [inline keras_nlp/models/task.py:194]: with a `backbone` argument

File: keras_nlp/models/task.py (hunk 2 above)
Comment (mattdangerw) [reply keras_nlp/models/task.py:190]: We should still flip the order here of this line. `raise ValueError` before `backbone = load_from_preset`.

Just a minor quality of life thing, but the `load_from_preset` call might download gigs of data and be slow, so if a user is trying to load with a custom backbone, we are better off error first, and sending the user on the right track (no need to download weights).

Comment (sampathweb) [reply keras_nlp/models/task.py]: Yes, that makes sense.  Done

File: keras_nlp/models/task.py (hunk 3 above)
Comment (sampathweb) [reply keras_nlp/models/task.py]: Typo fixed.

**Review by mattdangerw — APPROVED** [review]
lgtm!

**Review by nkovela1 — APPROVED** [review]
LGTM, thank you for the fix Ramesh!
